disdrodb/tests/test_l0a_processing.py

```diff
@@ -20,6 +29,190 @@
 )
 
 
+@pytest.fixture
+def create_dummy_config_file(request):
+    content = request.param[0]
+    file_name = request.param[1]
+    root_folder = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+    config_folder = os.path.join(root_folder, "l0", "configs")
+
+    test_folder = os.path.join(config_folder, "test")
+    if not os.path.exists(test_folder):
+        os.makedirs(test_folder)
+
+    test_file_path = os.path.join(test_folder, file_name)
+
+    with open(test_file_path, "w") as f:
+        print(test_file_path)
+        f.write(content)
+
+    yield
+    os.remove(test_file_path)
+    shutil.rmtree(test_folder)
+
+
+content = """key_1:

```

Comment:
Stuff can not be better encapsulated to avoid having these global variables? Something like the code here below (I am not sure it works ....)

```
def test_set_nan_unvalid_values() 
        content = "..." 
        other_var= "...."

        @pytest.mark.parametrize("create_dummy_config_file", [(content, file_name)], indirect=True)
        def test(create_dummy_config_file):
               # pass 

         test()
```

Reply:
This seems not to work :   


```
def test_set_nan_unvalid_values() : 
    content = """key_1:
    valid_values:
    - 1
    - 2
    - 3
    """
    file_name = "raw_data_format.yml"

    @pytest.mark.parametrize("create_dummy_config_file", [(content, file_name)], indirect=True)
    def test(create_dummy_config_file):
        df = pd.DataFrame({"key_1": [1, 2, 1, 2, 1]})
        output = set_nan_unvalid_values(df, "test", verbose=False)
        assert df.equals(output)

        # Test with modification
        df = pd.DataFrame({"key_1": [1, 2, 1, 2, 4]})
        output = set_nan_unvalid_values(df, "test", verbose=False)
        assert np.isnan(output["key_1"][4])
    
    test()
```
return a` TypeError: test_set_nan_unvalid_values.<locals>.test() missing 1 required positional argument: 'create_dummy_config_file'`


and 
```
def test_set_nan_unvalid_values() : 
    content = """key_1:
    valid_values:
    - 1
    - 2
    - 3
    """
    file_name = "raw_data_format.yml"

    @pytest.mark.parametrize("create_dummy_config_file", [(content, file_name)], indirect=True)
    def test(create_dummy_config_file):
        df = pd.DataFrame({"key_1": [1, 2, 1, 2, 1]})
        output = set_nan_unvalid_values(df, "test", verbose=False)
        assert df.equals(output)

        # Test with modification
        df = pd.DataFrame({"key_1": [1, 2, 1, 2, 4]})
        output = set_nan_unvalid_values(df, "test", verbose=False)
        assert np.isnan(output["key_1"][4])
    
    test(create_dummy_config_file)
```

This does not run the fuction` create_dummy_config_file()`

Reply:
@regislon and what happen (or not) if we do? 
```
def test_set_nan_unvalid_values(create_dummy_config_file):
    content = """key_1:
      valid_values:
      - 1
      - 2
      - 3
    """
    file_name = "raw_data_format.yml"
    # Test without modification
    df = pd.DataFrame({"key_1": [1, 2, 1, 2, 1]})
    output = set_nan_unvalid_values(df, "test", verbose=False)
    assert df.equals(output)
```